data/local_data_source/search/src/main/java/com/sanaa/search/search_history/dao/QueryDao.kt

```diff
@@ -0,0 +1,23 @@
+package com.sanaa.search.search_history.dao
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.sanaa.search.search_history.dto.QueryLocalDto
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface QueryDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertQuery(query: QueryLocalDto)
+
+    @Query("DELETE FROM queries WHERE search_query = :query")
+    suspend fun deleteQuery(query: String)
+
+    @Query("SELECT search_query FROM queries ORDER BY timestamp DESC LIMIT 10")
+    fun getAllQueries(): Flow<List<String>>
```

Comment:
Make the limit dynamic
```suggestion
    @Query("SELECT search_query FROM queries ORDER BY timestamp DESC LIMIT 10")
    fun getAllQueries(limit: Int): Flow<List<String>>
```